client/my-sites/exporter/exporter.jsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Gridicon from 'components/gridicon';
+import Button from 'components/forms/form-button';
+import AdvancedOptions from 'my-sites/exporter/advanced-options';
+
+export default React.createClass( {
+	displayName: 'Exporter',
+
+	render: function() {
+		return (
+			<div className="section-export">
+				<CompactCard>
+					<header>
+						<Button
+							className="exporter__export-button"
+							disabled={ false }
+							isPrimary={ true }
+						>
+							{ this.translate( 'Export' ) }
+						</Button>
+						<h1 className="exporter__title">
+							{ this.translate( 'Download an Export File' ) }
+						</h1>
+					</header>
+					<a href="#" onClick={ this.props.toggleAdvancedSettings }>
+						<Gridicon
+							icon={ this.props.advancedSettings.isVisible ? 'chevron-up' : 'chevron-down' }
+							size={ 16 } />
+						{ this.translate( 'Advanced Export Settings' ) }
+					</a>
+				</CompactCard>
+
+				{
+					this.props.advancedSettings.isVisible &&
+					<AdvancedOptions
```

Comment:
If I can add one more bit to the conversation, as having played with Redux and datastore for quite a while now, I think it's very critical that we weigh carefully the most pushed-for techniques on the web. It's not that I disagree with Dan Abramov or anything like that, but I've noticed often that there are very few projects on our scale with React, Redux, Flummox, Phlem, Phrogs, and Bogs. The separation between data components and visual components is incredibly important, but we also have to consider the other factors at our scale, one of which can be the proliferation of components.

From what I can gather on the wider Interwebs, most of what is said and discussed concerns the 95% of projects out there where more of the concern is not abusing these frameworks. We should follow 95% of their advice. In the other cases, we should have a clear warrant for not following them.

Two principles that I like to consider are how much data needs to be passed up and down the hierarchy and how significant any particular component will be.

For example, if we end up passing data down seven levels of components we have inseparably tied together all these components making a very fragile system full of data passing code when there's no need. On the other hand, if we have seven levels of components all watching the store we could run into performance issues as everything acts on its own and everything gets tied into the stores.

There's a difference, I think, in something like the exporter. How many instances of this particular component will ever be on display at once? Just a single instance - it's not like a button or a Card. Because of this, I think we can almost eliminate any performance concerns we might have because a single component won't make a dent against the issues something else might bring which on average is there in thirty different places across the app.

Generic settings widgets are great candidates for splitting off into UI-only components because they don't have any intrinsic ties to their data. On that note, they don't even need a data-gathering component because whatever form or module that displays the settings components should already have that data available - pure UI win. On the other hand, aren't specific-setting components already intrinsically tied to their data? Something like a `<WhatKindsOfThingsToExportSetting />` should only ever be used by the export by its very nature - not a big violation to get a little messier with the data ties.

The dogma will save us most of the time, but there's very very very little good information on the Internet on guidelines for apps our size (it's a great opportunity for us to learn and share!). Much more, it's significantly more difficult to filter out the noise since there are thousands of articles on Reactifluxiness.

Hope this helps!